app/scripts/modules/core/src/pipeline/config/templates/v2/configurePipelineTemplateModalV2.controller.ts

```diff
@@ -0,0 +1,251 @@
+      .then(() => {
+        this.groupVariableMetadata();
+        this.initializeVariables();
+        if (!this.variables.length) {
+          this.state.noVariables = true;
+        }
+      })
+      .then(() => (this.state.loading = false))
+      .catch(() => {
+        Object.assign(this.state, { loading: false, error: false, planErrors: null, loadingError: true });
+      });
+  }
+
+  public cancel(): void {
+    this.$uibModalInstance.dismiss();
+  }
+
+  public formIsValid(): boolean {
+    return this.variables.every(v => v.errors.length === 0);
+  }
+
+  public submit(): IPromise<void> {
+    const config = this.buildConfig();
+
+    return PipelineTemplateReader.getPipelinePlan(config)
+      .then(plan => {
+        const { parameterConfig, notifications, triggers } = plan;
+        const { inheritTemplateParameters, inheritTemplateNotifications, inheritTemplateTriggers } = this.state;
+
+        const configWithInheritedValues = {
+          ...config,
+          ...(inheritTemplateParameters && parameterConfig ? { parameterConfig } : {}),
+          ...(inheritTemplateNotifications && notifications ? { notifications } : {}),
+          ...(inheritTemplateTriggers && triggers ? { triggers } : {}),
+        };
+
+        this.$uibModalInstance.close({ plan, config: configWithInheritedValues });
+      })
+      .catch((response: IHttpPromiseCallbackArg<IPipelineTemplatePlanResponse>) => {
+        Object.assign(this.state, { loading: false, error: true, planErrors: response.data && response.data.errors });
+      });
+  }
+
+  public dismissError(): void {
+    Object.assign(this.state, { error: false, planErrors: null, loadingError: false });
+  }
+
+  public buildConfig(): IPipelineTemplateConfigV2 {
+    const {
+      application: { name: appName },
+      state: { inheritTemplateNotifications, inheritTemplateParameters, inheritTemplateTriggers },
+      pipelineName,
+      pipelineTemplateConfig,
+      source,
+    } = this;
+
+    const excludeConfig = [
+      ...(inheritTemplateParameters ? [] : ['parameters']),
+      ...(inheritTemplateNotifications ? [] : ['notifications']),
+      ...(inheritTemplateTriggers ? [] : ['triggers']),
+    ];
+
+    return {
+      ...(pipelineTemplateConfig || {}),
+      name: pipelineName,
+      application: appName,
+      variables: this.transformVariablesForPipelinePlan(),
+      exclude: excludeConfig,
+      ...PipelineTemplateV2Service.getPipelineTemplateConfigV2(source),
+    };
+  }
+
+  private loadTemplate(): IPromise<void> {
+    return PipelineTemplateReader.getPipelineTemplateFromSourceUrl(this.source, this.executionId, this.pipelineId).then(
+      template => {
+        this.template = template;
+      },
+    );
+  }
+
+  private transformVariablesForPipelinePlan(): { [key: string]: any } {
+    return chain(this.variables || [])
+      .cloneDeep()
+      .map(v => {
+        if (v.type === 'object') {
+          v.value = JSON.parse(v.value);
```

Comment:
YAML has been replaced with JSON for MPTv2

Reply:
I probably missed this but are there a defined set of possible types somewhere and are they just object, int, and float?

Reply:
Great question. There is MPTV2 documentation [here](https://www.spinnaker.io/guides/user/pipeline/pipeline-templates/) but I don't believe the variable types have been documented, looping in @jtk54 to confirm.